src/spec.rs

```diff
@@ -123,12 +281,81 @@ impl Zip64CentralDirectoryEndLocator {
         })
     }
 
-    pub fn write<T: Write>(&self, writer: &mut T) -> ZipResult<()> {
-        writer.write_u32_le(ZIP64_CENTRAL_DIRECTORY_END_LOCATOR_SIGNATURE)?;
-        writer.write_u32_le(self.disk_with_central_directory)?;
-        writer.write_u64_le(self.end_of_central_directory_offset)?;
-        writer.write_u32_le(self.number_of_disks)?;
-        Ok(())
+    pub fn block(self) -> Zip64CDELocatorBlock {
+        let Self {
+            disk_with_central_directory,
+            end_of_central_directory_offset,
+            number_of_disks,
+        } = self;
+        Zip64CDELocatorBlock {
+            magic: ZIP64_CENTRAL_DIRECTORY_END_LOCATOR_SIGNATURE,
+            disk_with_central_directory,
+            end_of_central_directory_offset,
+            number_of_disks,
+        }
```

Comment:
You may be able to use `..self` here.

Reply:
That looks like it should work, but I get this error!
```rust
error[E0308]: mismatched types
   --> src/spec.rs:287:15
    |
287 |             ..self
    |               ^^^^ expected `Zip64CDELocatorBlock`, found `Zip64CentralDirectoryEndLocator`s
```

It may be something to bring up to the lang team!